docs/user.md

```diff
@@ -0,0 +1,193 @@
+```
+
+##### Option B: Machine Set 
+
+If you have support for OpenShift's Machine API then you can create a new `MachineSet` to configure the labels and taints. See [OpenShift documentation](https://docs.openshift.com/container-platform/latest/machine_management/creating-infrastructure-machinesets.html#binding-infra-node-workloads-using-taints-tolerations_creating-infrastructure-machinesets) on how to configure a new `MachineSet`. Note that at time of certification testing, OpenShift's product documentation may not mention your infrastructure provider yet! 
+
+Here is a `MachineSet` YAML snippet on how to configure the label and taint as well:
+
+```yaml
+      metadata:
+        labels:
+          openshift-tests: "true"
+          node-role.kubernetes.io/tests: ""
+      taints:
+        - key: node-role.kubernetes.io/tests
+          effect: NoSchedule
+```
+
+
+### Privilege Requirements
+
+A user with [cluster administrator privilege](https://docs.openshift.com/container-platform/latest/authentication/using-rbac.html#creating-cluster-admin_using-rbac) must be used with the provider certification tool. You also use the default kubeadmin user if you wish. 
+
+
+## Install
+
+
+### Prebuilt Binary
+
+The provider certification tool is shipped as a single executable binary which can be downloaded from:
+
+[https://github.com/redhat-openshift-ecosystem/provider-certification-tool](https://github.com/redhat-openshift-ecosystem/provider-certification-tool)
+
+The provider certification tool can be used from any system with network access to the OpenShift cluster under test. 
+
+
+### Build from Source
+
+See the development guide for more information.
+*TODO - link*
+
+
+## Usage
+
+
+### Run provider certification tests
+
+```sh
+openshift-provider-cert run 
+openshift-provider-cert run -w # Keep watch open until completion
+```
+
+
+### Check status
+
+```sh
+openshift-provider-cert status 
+openshift-provider-cert status -w # Keep watch open until completion
+```
+
+
+### Retrieve the results
+
+The certification results must be retrieved from the OpenShift cluster under test using:
+
+```sh
+openshift-provider-cert retrieve
+openshift-provider-cert retrieve ./destination-dir/
+```
+
+You can see a summarized view of the results using:
+
+```sh
+openshift-provider-cert results retrieved-archive.tar.gz
+```
+
+
+### Environment Cleanup
+
+Once the certification process is complete and you are comfortable with destroying the test environment:
+
+```sh
+openshift-provider-cert destroy
+```
+
+You will need to destroy the OpenShift cluster under test separately. 
+
+
+### Certification Failures
+
+If you already know the reason for a certification failure then resolve the problem and re-run the provider certification tool again so a new certification archive is created. 
+
+If you are not sure why you have failed certification or if some of the tests fail intermittently, proceed with the troubleshooting steps below. 
+
+
+### Troubleshooting
+
```

Comment:
@bostrt new troubleshooting section exploring at the sonobuoy tarball, could you please add it?:

#### Result tarball file

The result tarball file follows the format of the backend used to run the certification environment: Sonobuoy.

Extract it to the `results` directory:

```bash
tar xfz <timestamp>_sonobuoy_<execution_id>.tar.gz -C results/
```

Once extracted, the tarball file is grouped in the following subdirectories:

```
results/
├── hosts
├── meta
├── plugins
├── podlogs
├── resources
├── servergroups.json
└── serverversion.json
```
- `hosts` provides the kubelet configuration and health check for each node on the cluster
- `meta` has the metadata collected from the cluster and certification environment
- `plugins` has the plugins definitions and results
- `podlogs` has the logs of pods used on the certification environment: server and plugins
- `resources` has all the manifests for all the resources cluster and namespace scoped.
- `servergroups.json` has the APIGroupList custom resource
- `serverversion.json` has the Kubernetes version

To start exploring the problems in the certification environment, you can start looking into the `podlogs` directory.

The file `results/plugins/<_plugin_name_>/sonobuoy_results.yaml` has the results for each test. If the test has been failed, you can see the reason on the field `.details.failure` and `.details.system-out`:

Using the `yq` tool you filter the failed tests by running this command:

- Getting the test names that have been `failed` from plugin `openshift-kube-conformance`:

```bash
yq -r '.items[].items[].items[] | select (.status=="failed") | .name ' results/plugins/openshift-kube-conformance/sonobuoy_results.yaml
```

- Get the `.failure` field for job `[sig-arch] Monitor cluster while tests execute`:

```bash
yq -r '.items[].items[].items[] | select (.name=="[sig-arch] Monitor cluster while tests execute").details.failure ' results/plugins/openshift-kube-conformance/sonobuoy_results.yaml
```